WordPress/Classes/ViewRelated/Me/All Domains/View Models/AllDomainsListViewModel.swift

```diff
@@ -8,24 +8,23 @@ class AllDomainsListViewModel {
     enum State {
         case normal
         case loading
-        case empty
-        case error
+        case empty(AllDomainsListEmptyStateViewModel)
```

Comment:
The reasoning behind this is that the 'error' state is not truly a distinct state on the All Domains screen. Instead, it is merely a triggering event that leads to the empty state.

The All Domains screen can transition to the `empty` state due to the following events:
- When the user has no domains.
- When an error occurs.
- When there are no domains that match the user's search query.

Reply:
"Empty" has two meanings in this file. The general empty state and the specific no domains (empty) empty state. To avoid confusion, I suggest renaming this enum case:
```suggestion
        case message(AllDomainsListEmptyStateViewModel)
```

or

```suggestion
        case fullscreenMessage(AllDomainsListEmptyStateViewModel)
```

And renaming `AllDomainsListEmptyStateViewModel` as well to remove the usage of "empty"

Reply:
What do you mean by "The general empty state" ? The "Empty" state can occur in 3 scenarios:

- The user doesn't have domains, meaning the All Domain endpoint returns an empty list.
- An error occurs.
- No domains match a search query.
All of these cases cause the list to be empty, hence the naming of "Empty" state.

Honestly I don't find it confusing. When the table view is empty, then you know the `viewModel.state == .empty`.

Reply:
> What do you mean by "The general empty state"?

`State.empty` now means either the list is empty or an error occurred, making the name `.empty` not 100% accurate. Yes, if an error occurs, the table is empty. But if I'm reading the code and read `.empty`, I'll assume there are no domains available, even though it could be a network issue for example.

There's currently a helper function `emptyStateViewModel` that creates the view model to pass to .empty when the domains list is empty. In this sentence I used "empty" twice, to imply two different but close things. This is where the confusion lies imo.

Reply:
Resolved in 4c86f70de1726cbdfc86414d784a8cc695c2987c